src/accountsdb/db.zig

```diff
@@ -3918,6 +4126,128 @@ test "shrink account file works" {
+        .geyser_writer = null,
+        .index_allocation = .ram,
+        .number_of_index_shards = ACCOUNT_INDEX_SHARDS,
+        .lru_size = null,
+    });
+    defer accounts_db.deinit();
+
+    // pretend `all_snapshot_fields`/`snap_files` refers to `tmp_snap_dir`, even though the archive file isn't actually in there, just the unpacked contents.
+    // TODO: this is not nice, make sure the API for loading from archives outside of the snapshot dir is improved.
+    _ = try accounts_db.loadWithDefaults(allocator, &all_snapshot_fields, 1, true, 1500);
+
+    var bank_fields = try BankFields.initRandom(allocator, random, 128);
+    defer bank_fields.deinit(allocator);
+
+    const full_slot = all_snapshot_fields.full.accounts_db_fields.slot;
+    const inc_slot = all_snapshot_fields.incremental.?.accounts_db_fields.slot;
+
+    const full_gen_result = try accounts_db.generateFullSnapshot(.{
+        .target_slot = full_slot,
+        .bank_fields = &bank_fields,
+        .lamports_per_signature = random.int(u64),
+        .old_snapshot_action = .ignore_old, // make sure we don't delete anything in `sig.TEST_DATA_DIR`
+        .deprecated_stored_meta_write_version = all_snapshot_fields.full.accounts_db_fields.stored_meta_write_version,
+    });
+
+    const inc_gen_result = try accounts_db.generateIncrementalSnapshot(.{
+        .target_slot = inc_slot,
+        .bank_fields = &bank_fields,
+        .lamports_per_signature = random.int(u64),
+        .old_snapshot_action = .ignore_old, // make sure we don't delete anything in `sig.TEST_DATA_DIR`
+        .deprecated_stored_meta_write_version = all_snapshot_fields.incremental.?.accounts_db_fields.stored_meta_write_version,
+    });
+
+    try std.testing.expectEqual(all_snapshot_fields.full.accounts_db_fields.bank_hash_info.accounts_hash, full_gen_result.hash);
+    try std.testing.expectEqual(all_snapshot_fields.full.bank_fields.capitalization, full_gen_result.capitalization);
+
+    try std.testing.expectEqual(all_snapshot_fields.incremental.?.bank_fields_inc.snapshot_persistence.?, inc_gen_result);
+
+    try std.testing.expectEqual(full_slot, inc_gen_result.full_slot);
+    try std.testing.expectEqual(full_gen_result.hash, inc_gen_result.full_hash);
+    try std.testing.expectEqual(full_gen_result.capitalization, inc_gen_result.full_capitalization);
+
+    const full_hash = full_gen_result.hash;
+    const inc_hash = inc_gen_result.incremental_hash;
+
+    const queue, var queue_lg = push_msg_queue_mux.readWithLock();
+    defer queue_lg.unlock();
+
+    try std.testing.expectEqual(2, queue.items.len);
+
+    const queue_item_0 = queue.items[0]; // should be from the full generation
+    const queue_item_1 = queue.items[1]; // should be from the incremental generation
+
+    // try std.testing.expect(try queue_item_0.verify(Pubkey.fromPublicKey(&my_keypair.public_key)));
+    // try std.testing.expect(try queue_item_1.verify(Pubkey.fromPublicKey(&my_keypair.public_key)));
```

Comment:
these probably shouldn't be here

Reply:
Good catch lol

Reply:
2d5918e47e425b93b6a62200e6adf1aa0759d81c